Convert two baremetal nodes to compute

brtkwr opened the pull request.

brtkwr committed:
Convert two baremetal nodes to compute

markgoddard reviewed (commented):
You'll need interface configuration in `etc/kayobe/inventory/group_vars/compute/network-interfaces`. See `etc/kayobe/inventory/group_vars/controllers/network-interfaces` for a starting point, but you will only need a subset of that.

You'll also need to update `etc/kayobe/compute.yml`, similar to `etc/kayobe/controllers.yml`. You'll want to add a role mapping for the computes in `etc/kayobe/overcloud.yml`. And you'll want to set `kolla_enable_neutron_provider_networks` to true.

brtkwr committed:
Config to add hypervisor

jovial commented on etc/kayobe/compute.yml:
```diff
@@ -5,6 +5,7 @@
 # User with which to access the computes via SSH during bootstrap, in order
 # to setup the Kayobe user account.
 #compute_bootstrap_user:
+compute_bootstrap_user: root
```
```suggestion
compute_bootstrap_user: centos
```
centos is default user for generic cloud image

brtkwr committed:
Update etc/kayobe/compute.yml

Co-authored-by: jovial <will@stackhpc.com>

brtkwr committed:
Add compute hosts to overcloud provisioning network

jovial committed:
Merge remote-tracking branch 'upstream/alaska-prod-add-hypervisors' into HEAD

Conflicts:
	etc/kayobe/inventory/hosts

jovial committed:
Fix kayobe network connectivity check

The following error was observed whilst runing kayobe network connectivity check:

failed: [sv-b16-u21] (item=alaska_ctl) => {
    "changed": false,
    "cmd": "ping  -c1 -M do  # localhost",
    "delta": "0:00:00.003493",
    "end": "2020-05-29 13:39:16.911225",
    "invocation": {
        "module_args": {
            "_raw_params": "ping  -c1 -M do  # localhost",

NOTE: I have addded the `#localhost` comment for debugging purposes it corresponds
to the variable, `remote_host`.

The problem was that the IPs for the seed hypervisor were not in network-allocations.
I've also renamed it to stop it being confused with the implicit local host.

This did not happen on every run as we use the random filter to pick a host.

jovial committed:
Add initial network configuration for compute hosts

jovial committed:
Adjust raid configuration

The nodes have 4 disks.

jovial committed:
Use overlay2 storage driver

This is so we can use the disks in a RAID-10 configuration.

jovial committed:
Update compute node IPs

The servers were reprovisioned and assigned different IPs.

jovial committed:
Convert to using a tagged interface for ilab

This is needed to boot VMs on the ilab network otherwise the tagged frames will
be dropped by the switch (VLAN 60 was the native vlan).

This requires you to reconfigure the switch mid host configure (or re-run after
failure) to reconfigure the switchport to allow tagged traffic on VLAN 60.

jovial committed:
Give hypervisors a more memorable alias

jovial committed:
Add playbook to enable nested virt

jovial committed:
Tweaks to nova.conf

- Enable nested virt
- Set overcommit

brtkwr committed:
Merge pull request #206 from jovial/feature/prod-compute

Add hypervisors to p3

brtkwr committed:
Update users

brtkwr committed:
Support Octavia

brtkwr committed:
Use wendy ssh key in passwords.yml

jovial commented on etc/kayobe/users.yml:
```diff
@@ -20,15 +21,8 @@ users_default:
-  - "{{ lookup('file', kayobe_config_path ~ '/ssh-keys/id_rsa_bharat.pub') }}"
-  - "{{ lookup('file', kayobe_config_path ~ '/ssh-keys/id_rsa_isaac.pub') }}"
-
-- username: isaac
```
Did you mean to remove @Wasaac's account definition? It is probably better practice not to share accounts. Also this won't delete the account.

brtkwr replied:
I thought it might have been added here by accident. Does it really make sense to have an account per user in all the overcloud and seed machines?

markgoddard replied:
Anyone who wants to run kayobe needs to have their key authorized for the stack and kolla users.

brtkwr replied:
The keys are intact for stack and kolla users

brtkwr committed:
Merge pull request #207 from SKA-ScienceDataProcessor/alaska-prod-octavia

Support Octavia

markgoddard commented on etc/kayobe/inventory/host_vars/ethsw-b16-u38:
```diff
@@ -146,6 +146,11 @@ switch_interface_config:
+    description: overcloud provisioning
+    config:
+      - "no shutdown"
+      - "untagged Te 1/4/4"
```
Why is this required?

jovial replied:
This is a relic from when we were going to move the nodes into bifrost. This was before we realised there was no physical connection between the 1G and 10G networks. We should remove this.

jovial committed:
Revert "Add compute hosts to overcloud provisioning network"

This reverts commit 473e21baf06fff528a25199d8bef5faa5f2226b8.

jovial reviewed (approved):
This is going to be the basis of the train config, could we merge so that the diff isn't so large? I've removed the contentious switch config.
